src/Mod/PartDesign/Gui/TaskTextParameters.cpp

```diff
@@ -0,0 +1,441 @@
+        // enter reference selection mode
+        TaskSketchBasedParameters::onSelectReference(true, true, false, true);
+    } else {
+        if (!pcText->getDocument()->isIn(lnk.getValue())){
+            Base::Console().Error("Object was deleted\n");
+            return;
+        }
+        propReferenceAxis->Paste(lnk);
+        exitSelectionMode();
+    }
+
+    try {
+        recomputeFeature();
+        updateUI();
+    }
+    catch (const Base::Exception& e) {
+        e.ReportException();
+    }
+}
+
+
+TaskTextParameters::~TaskTextParameters()
+{
+    try {
+        //hide the parts coordinate system axis for selection
+        PartDesign::Body * body = vp ? PartDesign::Body::findBodyOf(vp->getObject()) : 0;
+        if (body) {
+            App::Origin *origin = body->getOrigin();
+            ViewProviderOrigin* vpOrigin;
+            vpOrigin = static_cast<ViewProviderOrigin*>(Gui::Application::Instance->getViewProvider(origin));
+            vpOrigin->resetTemporaryVisibility();
+        }
+    } catch (const Base::Exception &ex) {
+        ex.ReportException();
+    }
+
+}
+
+void TaskTextParameters::changeEvent(QEvent *e)
+{
+    TaskBox::changeEvent(e);
+    if (e->type() == QEvent::LanguageChange) {
+        ui->retranslateUi(proxy);
+    }
+}
+
+void TaskTextParameters::getReferenceAxis(App::DocumentObject*& obj, std::vector<std::string>& sub) const
+{
+    if (axesInList.empty())
+        throw Base::RuntimeError("Not initialized!");
+
+    int num = ui->axis->currentIndex();
+    const App::PropertyLinkSub &lnk = *(axesInList[num]);
+    if (lnk.getValue() == 0) {
+        throw Base::RuntimeError("Still in reference selection mode; reference wasn't selected yet");
+    } else {
+        PartDesign::ProfileBased* pcRevolution = static_cast<PartDesign::ProfileBased*>(vp->getObject());
+        if (!pcRevolution->getDocument()->isIn(lnk.getValue())){
+            throw Base::RuntimeError("Object was deleted");
+        }
+
+        obj = lnk.getValue();
+        sub = lnk.getSubValues();
+    }
+}
+
+// this is used for logging the command fully when recording macros
+void TaskTextParameters::apply()
+{
+    std::vector<std::string> sub;
+    App::DocumentObject* obj;
+    getReferenceAxis(obj, sub);
+    std::string axis = buildLinkSingleSubPythonStr(obj, sub);
+    auto tobj = vp->getObject();
+    FCMD_OBJ_CMD(tobj,"ReferenceAxis = " << axis);
+    FCMD_OBJ_CMD(tobj,"TextString = \"" << propText->getValue() << "\"");
+    FCMD_OBJ_CMD(tobj,"Font = \"" << propFont->getValue() << "\"");
+    FCMD_OBJ_CMD(tobj,"Size = " << propSize->getValue());
+    FCMD_OBJ_CMD(tobj,"Height = " << propHeight->getValue());
+}
+
+
+// Based on https://stackoverflow.com/a/64729615
+// Ported to c++
+// Modified to ignore Non-TrueType fonts
+void TaskTextParameters::getFontPaths(void)
+{
+    QStringList font_paths = QStandardPaths::standardLocations(QStandardPaths::FontsLocation);
+    QFontDatabase* db = new QFontDatabase();
```

Comment:
Probably no reason to dynamically allocate this, is there?

Reply:
This was the best way I could figure out to map a font name to the actual file path that I need. Do you mean that I could precompute this somehow?

Anyway, don't spend too much time reviewing now, as the main functionality is not completed. I actually only made the PR now in order to discuss the segmentation fault. Mentioned above.

Reply:
I was just giving it a quick look for the obvious problem children (naked pointers being a major culprit in segfaults). What I mean is that there is no reason to create a pointer to the font database at all: just create the object on the stack:
```suggestion
    QFontDatabase db;
```
(and of course change your -> to . in later code...)

Reply:
Ah, of course- Done